src/libs/actions/IOU.js

```diff
@@ -1124,6 +1158,7 @@ function sendMoneyViaPaypal(report, amount, currency, comment, managerEmail, rec
 
     API.write('SendMoneyViaPaypal', params, {optimisticData, successData, failureData});
 
+    resetMoneyRequestInfo();
```

Comment:
I was trying to find a solution to this, but it doesn't going well. I'm thinking maybe we should make IOU onyx to a collection? Instead of holding one data only, we can make it hold each IOU id has its own onyx data.

```
key: iou_{iouType}{reportID}
```

This will allow us to initiate multiple money request (with different type and reportID) on different tabs. We don't need the `shouldReset` logic anymore too.

Reply:
This will work but seems somehow an overkill. Any other options?

Reply:
@bernhardoj I'm afraid the last commit adds more complexity for not much value. Do you think maybe we can store the current step in onyx and just navigate to that step in case we start a money request from another tab? or any other more straightforward solution

Reply:
I don't think that will solve the issue. The issue we are trying to solve is the different type of request is initiated. Before the latest commit, there are quite a lot of bugs when doing 2 requests in 2 tabs. It is buggy because of the reset logic. I wish we can use context 😭.

Reply:
Btw, here is a "crazy" idea I guess to avoid Onyx and Context, passing the value to params.
```
http://localhost:8080/send/new?currency=ILS&amount=234&logins=a@gmail.com,b@gmail.com&description=eangaegoneagao
http://localhost:8080/send/new?currency=ILS&amount=234&reportID=123456&isPolicyExpenseChat=true&description=eangaegoneagao
```

Reply:
Please no :laughing: URL parameters are a pain, we do it for the `backTo` and it's not really that great let's avoid that.

The iou collection solution is not looking that bad after all :sweat_smile:

Reply:
@bernhardoj Any plans here? (assuming we will revert the latest commit)

Reply:
I think the only plan we have left is to make it as a collection. But actually, there are 2 things that I don't like making it as a collection:
1. There will be lot of money request data stored in onyx
2. I think it may confuse user when they go to /send/new, /request/new, /send/new/12345 and see all different data
But this is the "best" thing we have for now.

Reply:
Okay, let's take a step back and rethink this without doing much logic
1. Revert last commit (except the part where `shouldEnforceFallback` is now `true`)
2. When going to TAB 2 and starting a new money request, the onyx data should be reset right?
3. So why on TAB 1 we are still seeing the same participants? I would assume the participants to get empty

Reply:
1. I think let's find a solution first before reverting it.
2. Yes
3. Yes, it's empty. I'm assuming you are talking about this issue https://github.com/Expensify/App/pull/17964#issuecomment-1584422120.

History:
1. We have the above issue, so the reset logic is added
2. The reset logic is buggy when doing a request in multiple tabs, so I added the last commit

Reply:
> What will happen?

Reset the iou since the id is different from what we have in onyx and navigate to the first page. This logic can be kept. but the other logic that set the id to a different id (instead of resetting) is a little overhead (the one that was added in the last commit)

Reply:
First of all, it's my mistake answering the 3rd question as Yes. Before the last commit, we have empty deps array in useEffect. So, nothing will happen when the participants or other data becomes empty.

> Reset the iou since the id is different from what we have in onyx and navigate to the first page

Ok, so this is clear.

> but the other logic that set the id to a different id (instead of resetting) is a little overhead (the one that was added in the last commit)

It is to cover all cases I posted it here https://github.com/Expensify/App/pull/17964#issuecomment-1588849643. You can revert locally and test those cases and you will notice the bug. Here is one of the bug with the old logic (with a difference of the deps array):
```
    useEffect(() => {
        const moneyRequestId = `${iouType.current}${reportID.current}`;
        const shouldReset = props.iou.id !== moneyRequestId;
        if (shouldReset) {
            IOU.resetMoneyRequestInfo(moneyRequestId);
        }
        if (_.isEmpty(props.iou.participants) || props.iou.amount === 0 || shouldReset) {
            Navigation.goBack(ROUTES.getMoneyRequestRoute(iouType.current, reportID.current), shouldReset);
        }
    }, [props.iou.participants, props.iou.amount, props.iou.id]);
```

TAB 1: request, TAB 2: send
1. In TAB 1, user start a REQUEST and proceed to confirm page
ID: request
2. In TAB 2, user start a SEND
What will happen? 
In TAB 1, IOU.startMoneyRequest -> resetMoneyRequestInfo(send), ID: send
In TAB 2, the useEffect rerun, resetMoneyRequestInfo(request) is called, ID: request
final ID: request
3. In TAB 2, enter any amount and press Next
When participant page is mounted, the useEffect logic above is executed. Remember, the current ID is request. Because the current ID is request and the URL is /send/new/participant, this means we will once again call resetMoneyRequestInfo(send) and be navigated back to amount page.
Guess what happen in TAB 1? componentDidUpdate is triggered because the ID is updated to "send", and because it's different from the URL (/request/new), resetMoneyRequestInfo(request) is called again and this will happen indefinitely.